config/tests/samples/create/samples.go

```diff
@@ -236,65 +235,85 @@ func waitForDeleteToComplete(t *Harness, wg *sync.WaitGroup, u *unstructured.Uns
 	}
 }
 
-// LoadSamples loads all the samples
-func LoadSamples(t *testing.T, project testgcp.GCPProject) []Sample {
+// LoadAllSamples loads all the samples.
+func LoadAllSamples(t *testing.T, project testgcp.GCPProject) []Sample {
 	matchEverything := regexp.MustCompile(".*")
-	return loadSamplesOntoUnstructs(t, matchEverything, project)
+	return LoadMatchingSamples(t, matchEverything, project)
 }
 
-func loadSamplesOntoUnstructs(t *testing.T, regex *regexp.Regexp, project testgcp.GCPProject) []Sample {
+// LoadMatchingSamples loads the samples that match the regex
+func LoadMatchingSamples(t *testing.T, regex *regexp.Regexp, project testgcp.GCPProject) []Sample {
+	sampleKeys := ListMatchingSamples(t, regex)
+	var samples []Sample
+	for _, sampleKey := range sampleKeys {
+		sample := loadSampleOntoUnstructs(t, sampleKey, project)
+		samples = append(samples, sample)
+	}
+	return samples
+}
+
+// ListAllSamples gets the keys for all the samples without loading them.
+func ListAllSamples(t *testing.T) []SampleKey {
+	matchEverything := regexp.MustCompile(".*")
+	return ListMatchingSamples(t, matchEverything)
+}
+
+// LoadSample loads one sample
+func LoadSample(t *testing.T, sampleKey SampleKey, project testgcp.GCPProject) Sample {
+	return loadSampleOntoUnstructs(t, sampleKey, project)
+}
+
+// SampleKey contains the metadata for a sample.
+// This lets us defer variable substitution.
+type SampleKey struct {
+	Name  string
+	files []string
+}
+
+func loadSampleOntoUnstructs(t *testing.T, sampleKey SampleKey, project testgcp.GCPProject) Sample {
 	t.Helper()
 
-	samples := make([]Sample, 0)
-	sampleNamesToFiles := mapSampleNamesToFilePaths(t, regex)
 	subVars := newSubstitutionVariables(t, project)
-	for sample, files := range sampleNamesToFiles {
-		resources := make([]*unstructured.Unstructured, 0)
-		for _, f := range files {
-			unstructs := readFileToUnstructs(t, f, subVars)
-			resources = append(resources, unstructs...)
-		}
-		s := Sample{
-			Name:      sample,
-			Resources: resources,
-		}
-		samples = append(samples, s)
+	resources := make([]*unstructured.Unstructured, 0)
+	for _, f := range sampleKey.files {
+		unstructs := readFileToUnstructs(t, f, subVars)
+		resources = append(resources, unstructs...)
 	}
-	return samples
+	s := Sample{
+		Name:      sampleKey.Name,
+		Resources: resources,
+	}
+	return s
 }
 
-func mapSampleNamesToFilePaths(t *testing.T, regex *regexp.Regexp) map[string][]string {
+// ListMatchingSamples gets the keys for all samples matching the regex, without loading them.
```

Comment:
Error: vet: config/tests/samples/create/contents_test.go:32:13: undefined: mapSampleNamesToFilePaths

Reply:
Will hopefully be fixed by rebase!